tasks/DroneFlightControlTask.cpp

```diff
@@ -45,26 +46,90 @@ bool DroneFlightControlTask::startHook()
         return false;
 
     mLastMission = Mission();
-
-    mAuthorityStatus = mVehicle->obtainCtrlAuthority(mFunctionTimeout);
-    ACK::getErrorCodeMessage(mAuthorityStatus, __func__);
     return true;
 }
 
-typedef DroneFlightControlTask::States TaskState;
-static TaskState djiStatusFlightToTaskState(uint8_t status)
+DroneFlightControlTask::States
+DroneFlightControlTask::runtimeStatesTransition(Telemetry::SDKInfo control_status)
 {
-    switch (status)
+    DroneFlightControlTask::States current_state = state();
+    switch (current_state)
     {
-        case DJI::OSDK::VehicleStatus::FlightStatus::STOPED:
-            return TaskState::DJI_STOPPED;
-        case DJI::OSDK::VehicleStatus::FlightStatus::ON_GROUND:
-            return TaskState::DJI_ON_GROUND;
-        case DJI::OSDK::VehicleStatus::FlightStatus::IN_AIR:
-            return TaskState::DJI_IN_AIR;
+        case RUNNING:
+        {
+            return TELEMETRY;
+        }
+        case TELEMETRY:
+        {
+            if (_cmd_action.connected())
+            {
+                return CONTROLLING;
+            }
+            else
+            {
+                return TELEMETRY;
+            }
+        }
+        case CONTROLLING:
+        {
+            bool can_take_ctrl = canTakeControl(control_status);
+            if (can_take_ctrl)
+            {
+                return CONTROLLING;
+            }
+            else if (!_cmd_action.connected())
+            {
+                return TELEMETRY;
+            }
+            else
+            {
+                return CONTROL_LOST;
+            }
+        }
+        case CONTROL_LOST:
+        {
+            if (_cmd_action.connected())
+            {
+                return CONTROL_LOST;
+            }
+            else
+            {
+                return TELEMETRY;
+            }
+        }
+        default:
+            return current_state;
+    }
+}
+
+void DroneFlightControlTask::applyTransition(
+    DroneFlightControlTask::States const& next_state)
+{
+    switch (next_state)
+    {
+        case CONTROLLING:
+        {
+            Time timeout = Time::now() + Time::fromSeconds((double)(mFunctionTimeout));
```

Comment:
Not sure it's true in english, but I tend to call this a deadline to avoid confusion between the timepoint (absolute, deadline) and the duration (timeout).